app/application/services/verification_service.py

```
@@ -104,10 +104,59 @@ async def verify_subject_chain(
+                )
+                all_events.extend(batch)
+                if len(batch) < batch_size:
+                    break
+                offset += batch_size
+            return await self._verify_tenant_events(tenant_id, all_events)
+
+        try:
+            if self.timeout_seconds is not None:
+                return await asyncio.wait_for(_run(), timeout=self.timeout_seconds)
+            return await _run()
+        except asyncio.TimeoutError:
```

Ruff (0.15.0)

[warning] Replace aliased errors with `TimeoutError`

Replace `asyncio.TimeoutError` with builtin `TimeoutError`

(UP041)